pyproject.toml

```diff
@@ -0,0 +1,40 @@
+[build-system]
+requires = [
+    "setuptools",
+    "wheel"
+]
+build-backend = "setuptools.build_meta"
+
+[tool.pytest.ini_options]
+addopts = "-v --tb=native --server=mysql://ensadmin:$ENSADMIN_PSW@mysql-ens-compara-prod-1:4485/"
+testpaths = [
+    "src/python/tests"
+]
+norecursedirs = [
+    "*.egg*",
+    "data",
+    "docs",
+    "env",
+    "tmp*"
+]
+
+[tool.coverage.run]
+source = ["src/python/lib"]
+
+[tool.mypy]
+ignore_missing_imports = true
+warn_unused_configs = true
+
+[tool.pylint.messages_control]
+max-line-length = 110
+disable = [
+    "invalid-name",
+    "no-self-use",
+    "relative-import",
+    "similarities",
+    "too-few-public-methods",
+    "too-many-arguments",
+    "too-many-locals",
+    "too-many-statements",
+    "wildcard-import"
```

Comment:
I would need a really compelling case to disable these two since `with` is strongly suggested by Python as it handles the handler closure internally, and the `f-string` provides more readable formatting of strings and it has been proven to have better performance.

Reply:
There are 2 cases listed in the `feature/homology_annotation` branch for both ([here](https://github.com/Ensembl/ensembl-compara/blob/feature/homology_annotation/src/python/lib/ensembl/compara/runnable/SplitFasta.py#L97) and [here](https://github.com/Ensembl/ensembl-compara/blob/feature/homology_annotation/src/python/lib/ensembl/compara/runnable/SplitFasta.py#L102)). But on another note, I think with some vigilance, these can be picked up by the reviewer's eye if necessary, and in the team we can push for the usage.

Reply:
I still think there are situations in which `consider-using-with` should not be enforced.

Reply:
So far Travis seems happy with this. When these come I will be happy to discuss it again :smile: